src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_POSITION = 0;
+    private static final int HOURS_POSITION = 2;
+    private static final int SALARY_POSITION = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+        LocalDate dayFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate dayTo = LocalDate.parse(dateTo, FORMATTER);
+        for (String name : names) {
+            int counter = 0;
+            for (String datum : data) {
+                String[] parts = datum.split(" ");
+                if (!datum.contains(name)) {
```

Comment:
How such solution will work if we have "John Doe" and "John Doe Jr." in report?
Seems someone will be overpaid :)